Dockerfile

```diff
@@ -0,0 +1,66 @@
+FROM i386/ubuntu:bionic as base_config
+ENV BYOND_MAJOR=512 \
+    BYOND_MINOR=1488 \
+    RUST_G_VERSION=0.2.0
+    
+
+FROM base_config as byond
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    curl \
+    unzip \
+    make \
+    && curl "http://www.byond.com/download/build/${BYOND_MAJOR}/${BYOND_MAJOR}.${BYOND_MINOR}_byond_linux.zip" -o byond.zip \
+    && unzip byond.zip \
+    && cd byond \
+    && sed -i 's|install:|&\n\tmkdir -p $(MAN_DIR)/man6|' Makefile \
+    && make install \
+    && chmod 644 /usr/local/byond/man/man6/* \
+    && apt-get purge -y --auto-remove curl unzip make \
+    && cd .. \
+    && rm -rf byond byond.zip
+
+
+FROM base_config as rust_g
+WORKDIR /rust_g
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    git \
+    ca-certificates \
+    libssl-dev \
+    pkg-config \
+    curl \
+    gcc-multilib \
+    && curl https://sh.rustup.rs -sSf | sh -s -- -y \
+    && git init \
+    && git remote add origin https://github.com/ParadiseSS13/rust-g
+RUN git fetch --depth 1 origin "${RUST_G_VERSION}" \
+    && git checkout FETCH_HEAD \
+    && ~/.cargo/bin/cargo build --release
+
+
+
+FROM base_config as paradise_build
+WORKDIR /paradise
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    libmariadb3 \
+    libssl1.0.0
+RUN apt-get autoremove
+RUN rm -rf /var/lib/apt/lists/*
+RUN cp /usr/lib/i386-linux-gnu/libmariadb.so.3 /usr/lib/libmariadb.so

```

Comment:
We dont use libmariaDB, we still on the old libmysql, and I am not sure on full compatibility

Reply:
tgstation seems to be running mariadb without issue, and given the number of similarities I suspect compatibility will not be an issue. I have worked on mysql -> mariadb conversions previously on large db instances and have never had significant incompatibility issues. 

Having said that I have only performed limited testing. Given that this contribution does not impact the existing functionality of the Paradise code base I would suggest leaving it in place. If users decide to use this experience issues then that may aid any future mariadb migration of the existing code base.

If the reasons above are inadequate I will replace this with the mysql library.

Reply:
TG runs mariaDB because they require it and are also running the BSQL library with the `SSdbcore` subsystem. We have the `SSdbcore` subsystem in development, but BSQL is not in the pipeline as far as I know. If the `libmariadb.so` is a drop in replacement library for `libmysql` then thats ok, but ideally we want to stick to what we already have. 

It doesnt affect migration in the slightest due to this only being game side, not server side. To me, using `libmysql.so` would be a better solution as its compatible with both MySQL and MariaDB, and our windows library is still using `libmysql.dll`

Reply:
I'm working on this now, and will open a new PR based on your feedback below, however it seems that byond (latest version 512.1488) is actively looking for libmariadb.so. If I install libmysql.so it fails with a message 'cannot find libmariadb.so'.

I can work around it, by renaming libmysql.so to libmariadb.so but this feels especially hacky. Do you still wish to proceed with using libmysql.so?

Reply:
I did some research (and painful reverse engineering), and the DLL support is fully BYOND side, and were using a DB framework from 2002. If `libmariadb.so` works under linux, then use that.